File: chap03/frontend/models/lib/block_config.dart
class BlockConfig {
  final double? horizontalPadding;
  final double? verticalPadding;
  final double? blockHeight;
  final double? horozontalSpacing;
  final double? blockWidth;

  const BlockConfig({
    this.horizontalPadding,
    this.verticalPadding,
    this.blockHeight,
    this.horozontalSpacing,
    this.blockWidth,
  });

  factory BlockConfig.fromJson(Map<String, dynamic> json) {
    return BlockConfig(
      horizontalPadding: json['horizontalPadding'] as double?,
      verticalPadding: json['verticalPadding'] as double?,
      blockHeight: json['blockHeight'] as double?,
      horozontalSpacing: json['horozontalSpacing'] as double?,
      blockWidth: json['blockWidth'] as double?,
    );
  }

  BlockConfig withRatio(double ratio) {
    return BlockConfig(
      horizontalPadding: (horizontalPadding ?? 0) * ratio,
      verticalPadding: (verticalPadding ?? 0) * ratio,
      blockHeight: (blockHeight ?? 0) * ratio,
      horozontalSpacing: (horozontalSpacing ?? 0) * ratio,
      blockWidth: (blockWidth ?? 0) * ratio,
    );
  }
}

File: chap06/backend/src/main/java/com/mooc/backend/entities/BlockConfig.java
package com.mooc.backend.entities;

import lombok.Data;

import java.io.Serial;
import java.io.Serializable;

@Data
public class BlockConfig implements Serializable {
    @Serial
    private static final long serialVersionUID = 1L;

    private Double horizontalPadding;

    private Double verticalPadding;

    private Double horizontalSpacing;

    private Double verticalSpacing;

    private Double blockWidth;

    private Double blockHeight;

    private String backgroundColor;

    private String borderColor;

    private Double borderWidth;
}

